hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -462,6 +462,11 @@ object DataSourceWriteOptions {
     .defaultValue(false)
     .withDocumentation("Whether to sync the table as managed table.")
 
+  val HIVE_BATCH_SYNC_PARTITION_NUM: ConfigProperty[String] = ConfigProperty
```

Comment:
`ConfigProperty[String]` -> `ConfigProperty[Integer]`

Reply:
oh, i see many boolean type config property is set to be ConfigProperty[String],   so i follow this style, make this config to String

in scala, it's better to set it Int instead of Integer.     fix the comment.